src/request/mod.rs

```diff
@@ -1,105 +1,42 @@
 //! RobTop's eccentricities are hidden. This is since directly re-using these structs outside of
 //! making/proxying requests for the boomlings servers seems rather useless to me, as they already
 //! contain a lot of boomlings-specific fields.
+//! This can also be edited for a specific GDPS, e.g 1.9 GDPS. (hi absowute :3)
 
 use crate::{model::GameVersion, serde::RequestSerializer};
 use serde::{Deserialize, Serialize};
+use once_cell::sync::OnceCell;
 
-macro_rules! const_setter {
-    ($name: ident, $field: ident, $t: ty) => {
-        pub const fn $name(mut self, $field: $t) -> Self {
-            self.$field = $field;
-            self
-        }
-    };
-
-    ($name: ident, $t: ty) => {
-        pub const fn $name(mut self, arg0: $t) -> Self {
-            self.$name = arg0;
-            self
-        }
-    };
+static GDPS_URL: OnceCell<String> = OnceCell::new();
 
-    ($(#[$attr:meta])* $name: ident: $t: ty) => {
-        $(#[$attr])*
-        pub const fn $name(mut self, $name: $t) -> Self {
-            self.$name = $name;
-            self
-        }
-    };
-
-    ($(#[$attr:meta])* $field:ident[$name: ident]: $t: ty) => {
-        $(#[$attr])*
-        pub const fn $name(mut self, $field: $t) -> Self {
-            self.$field = $field;
-            self
-        }
-    }
+pub fn get_gdps_url() -> String {
+    GDPS_URL.get_or_init(|| std::env::var("GDPS_URL").unwrap_or_else(|_| "https://www.boomlings.com/database/".to_string()))
 }
 
-pub mod comment;
-pub mod level;
-pub mod user;
-
-pub const REQUEST_BASE_URL: &str = "https://www.boomlings.com/database/";
-
-/// A `BaseRequest` instance that has all its fields set to the
-/// same values a Geometry Dash 2.1 client would use
 pub const GD_21: BaseRequest = BaseRequest::new(
     GameVersion::Version { major: 2, minor: 1 },
     GameVersion::Version { major: 3, minor: 3 },
     "Wmfd2893gb7",
 );
 
-/// A `BaseRequest` instance that has all its fields set to the
-/// same values a Geometry Dash 2.2 client would use
 pub const GD_22: BaseRequest = BaseRequest::new(
     GameVersion::Version { major: 2, minor: 2 },
     GameVersion::Version { major: 3, minor: 8 },
     "Wmfd2893gb7",
 );
 
-/// Base data included in every request made
-///
-/// The fields in this struct are only relevant when making a request to the
-/// `boomlings` servers. When using GDCF with a custom Geometry Dash API, they
-/// can safely be ignored.
 #[derive(Debug, Clone, Hash, Copy, PartialEq, Eq, Serialize, Deserialize)]
 pub struct BaseRequest<'a> {
-    /// The version of the game client we're pretending to be
-    ///
-    /// ## GD Internals:
-    /// This field is called `gameVersion` in the boomlings API and needs to be
-    /// converted to a string response
-    /// The value of this field doesn't matter, and the request will succeed
-    /// regardless of what it's been set to
-    #[serde(rename = "gameVersion")]
```

Comment:
please revert the deletion of all this as well (all all other comments that were deleted)